library/src/commonMain/kotlin/com/lagradost/cloudstream3/extractors/SobreatsesuypExtractor.kt

```diff
@@ -0,0 +1,73 @@
+// ! Bu araç @keyiflerolsun tarafından | @KekikAkademi için yazılmıştır.
+
+package com.lagradost.cloudstream3.extractors
+
+import android.util.Log
+import com.lagradost.cloudstream3.*
+import com.lagradost.cloudstream3.utils.*
+import com.fasterxml.jackson.annotation.JsonProperty
+
+open class Sobreatsesuyp : ExtractorApi() {
+    override val name            = "Sobreatsesuyp"
+    override val mainUrl         = "https://sobreatsesuyp.com"
+    override val requiresReferer = true
+
+    override suspend fun getUrl(url: String, referer: String?, subtitleCallback: (SubtitleFile) -> Unit, callback: (ExtractorLink) -> Unit) {
+        val ext_ref = referer ?: ""
+
+        val video_req = app.get(url, referer=ext_ref).text
+
+        val file     = Regex("""file\":\"([^\"]+)""").find(video_req)?.groupValues?.get(1) ?: throw ErrorLoadingException("File not found")
+        val postLink = "${mainUrl}/" + file.replace("\\", "")
+        val rawList  = app.post(postLink, referer=ext_ref).parsedSafe<List<Any>>() ?: throw ErrorLoadingException("Post link not found")
+
+        val postJson: List<SobreatsesuypVideoData> = rawList.drop(1).map { item ->
+            val mapItem = item as Map<*, *>
+            SobreatsesuypVideoData(
+                title = mapItem["title"] as? String,
+                file  = mapItem["file"]  as? String
+            )
+        }
+        Log.d("Kekik_${this.name}", "postJson » ${postJson}")
+
+		val vid_links = mutableSetOf<String>()
+        val vid_map   = mutableListOf<Map<String, String>>()
+        for (item in postJson) {
+            if (item.file == null || item.title == null) continue
+
+            val fileUrl   = "${mainUrl}/playlist/" + item.file.substring(1) + ".txt"
+            val videoData = app.post(fileUrl, referer=ext_ref).text
+
+			if (videoData in vid_links) { continue }
```

Comment:
This should already be built in https://github.com/recloudstream/cloudstream/blob/master/app/src/main/java/com/lagradost/cloudstream3/ui/player/RepoLinkGenerator.kt#L150

Reply:
I don't understand this

Reply:
Cs3 already removes duplicate links, so you dont have to do that check too